lib/rides/pages/search_ride_page.dart

```diff
@@ -125,133 +105,270 @@ class _SearchRideFormState extends State<SearchRideForm> {
+              _startSuggestion = suggestion;
+              loadRides();
+            }),
+            onDestinationSelected: (AddressSuggestion suggestion) => setState(() {
+              _destinationSuggestion = suggestion;
+              loadRides();
+            }),
           ),
         ),
-      );
-    }
+        IconButton(
+          onPressed: () => setState(() {
+            String oldStartText = _startController.text;
+            _startController.text = _destinationController.text;
+            _destinationController.text = oldStartText;
+            AddressSuggestion? oldStartSuggestion = _startSuggestion;
+            _startSuggestion = _destinationSuggestion;
+            _destinationSuggestion = oldStartSuggestion;
+            loadRides();
+          }),
+          icon: const Icon(Icons.swap_vert),
+        )
+      ],
+    );
   }
 
   Widget buildDatePicker() {
     _dateController.text = localeManager.formatDate(_selectedDate);
 
-    return Expanded(
-      child: Semantics(
-        button: true,
-        child: TextFormField(
-          decoration: InputDecoration(
-            border: const OutlineInputBorder(),
-            labelText: S.of(context).formDate,
-          ),
-          readOnly: true,
-          onTap: _showDatePicker,
-          controller: _dateController,
+    return Semantics(
+      button: true,
+      child: TextFormField(
+        decoration: InputDecoration(
+          border: const OutlineInputBorder(),
+          labelText: S.of(context).formDate,
         ),
+        readOnly: true,
+        onTap: _showDatePicker,
+        controller: _dateController,
       ),
     );
   }
 
   Widget buildTimePicker() {
     _timeController.text = localeManager.formatTime(_selectedDate);
 
-    return Expanded(
-      child: Semantics(
-        button: true,
-        child: TextFormField(
-          decoration: InputDecoration(
-            border: const OutlineInputBorder(),
-            labelText: S.of(context).formTime,
-          ),
-          readOnly: true,
-          onTap: _showTimePicker,
-          controller: _timeController,
-          validator: _timeValidator,
+    return Semantics(
+      button: true,
+      child: TextFormField(
+        decoration: InputDecoration(
+          border: const OutlineInputBorder(),
+          labelText: S.of(context).formTime,
+          errorText: _dateTimeValidator(_timeController.text),
         ),
+        readOnly: true,
+        onTap: _showTimePicker,
+        controller: _timeController,
+        validator: _dateTimeValidator,
       ),
     );
   }
 
-  Widget buildSeatsPicker() {
-    return Expanded(
+  Widget buildMainContentSliver() {
+    if (_rideSuggestions == null || _loading) {
+      if (_startSuggestion == null || _destinationSuggestion == null) {
+        return SliverToBoxAdapter(
+          child: Column(
+            children: <Widget>[
+              Image.asset('assets/pointing_up.png'),
```

Comment:
I think its better without the Emojis. Just the nice test telling you what to do in the center

Reply:
I thought we should use little cute pictures for every one of these kind of "empty" messages, since an empty page with only text would look very bland I suppose

Reply:
I like the emoji too, it definitely fits with the style we have used on other pages